Versioner/PrinceOfVersions.swift

```diff
@@ -103,5 +119,69 @@ public struct PrinceOfVersions {
             }
         })
     }
+}
 
+extension PrinceOfVersions: URLSessionDelegate {
+    
+    public func urlSession(_ session: URLSession, didReceive challenge: URLAuthenticationChallenge, completionHandler: @escaping (URLSession.AuthChallengeDisposition, URLCredential?) -> Void) {
+        
+        guard let trust = challenge.protectionSpace.serverTrust, SecTrustGetCertificateCount(trust) > 0 else {
+            completionHandler(.cancelAuthenticationChallenge, nil)
+            return
+        }
+        
+        guard _pinnedCertificateURL != nil else {
+            completionHandler(.performDefaultHandling, nil)
+            return
+        }
+        
+        if let serverCertificate = SecTrustGetCertificateAtIndex(trust, 0), let serverCertificateKey = publicKey(for: serverCertificate) {
+            
+            let hasKey = pinnedKeys().contains(where: { (key) -> Bool in
+                if (serverCertificateKey as AnyObject).isEqual(key) {
+                    return true
+                } else {
+                    return false
+                }
+            })
+            
+            if hasKey {
+                completionHandler(.useCredential, URLCredential(trust: trust))
+                return
+            }
+        }
+        
+        completionHandler(.cancelAuthenticationChallenge, nil)
+    }
+    
+    fileprivate func pinnedKeys() -> [SecKey] {
```

Comment:
Ako dohvacate cert koji se pina online, onda nema smisla pinnat cert, buduci da nije dosao sigurnim kanalom. Cert mora ili biti vec dodan u app, ili pinnanje nema smisla, tako da mi ova metoda ne radi nista posebno. Full je lagano podvaliti fejk cert, buduci da cemo ga ionako skinuti pa kao pinnati.

Reply:
Hm, nisam te bas shvation na sto mislis. Ne dohvaca se cert online pa se onda postavlja za pinnanje. URL koji se predaje je lokalni. Znaci nesto ovako: `Bundle.main.url(forResource: "certificate", withExtension: "crt")`

Reply:
Ohhh. Bilo mi je cudno jer izgleda kao da ga dohvacas remotely, sto je isto teoretski moguce. Jel PoV pod ili? Jer mozes procitat sve certove iz nekog bundlea, tako to radi Alamofire.

Reply:
Implementirano iteriranje kroz sve certifikate u main bundle i pinnanje svih keyeva (kao kod Alamofirea). Also, updetao sam README file sa sampleom za SecCert usage.